daemon/graphdriver/overlay2/overlay.go

```diff
@@ -234,9 +234,6 @@ func parseOptions(options []string) (*overlayOptions, error) {
 		}
 		key = strings.ToLower(key)
 		switch key {
-		case "overlay2.override_kernel_check":
-			// TODO(thaJeztah): change this to an error, see https://github.com/docker/cli/pull/3806
```

Comment:
Should we change this to an error, as your comment suggests? :sweat_smile:

Reply:
Yeah, I was on the fence; do we need a specific error, or is the `fmt.Errorf("overlay2: unknown option %s", key)` good enough?

I had some variation of the deprecation message at first (but then we would need another TODO to remove it (?) 😂 

Happy to hear what you (and others) think.

Reply:
Oh, we already have a fallback error; sorry I missed that!

Reply:
Yes, it's definitely less "nice" than a dedicated error, but considering the options, we were already printing a warning (so it _shouldn't_ come as a surprise), and I _doubt_ there's enough users of this option to warrant adding a special case, so KISS 😅